uphold/vault-assist-tool
# Fix spanish language detection not working

This fixes the Spanish language detection for translations. There was some refactoring of the i18n implementation but from my testing I could not find any issues. Even though I couldn't find any, it is a good idea to do another run through the app to make sure it's working correctly (for anything that doesn't, it would show the i18n key like `transaction.navigation.title` instead of `Vault Assist`. 

Also I removed the dayjs dependency, since I missed that in the previous dependency cleanup PR. 

## How to add Spanish language entries
This PR does not include the changes needed to add the `es` locale strings, to do so you should add the following:

1. Create the translation file in `src/assets/locales/es.js` so you can add the entries for translation:
```js
/* eslint-disable prettier/prettier */
export const es = {
  translation: {

  }
};
```

2. Add the export to `src/assets/locales/index.js`
```js
export * from './es';
```

## How to test
It is important to know that i18next stores a cache of the language it detects from the browser, so **when you change languages** in your OS you will need to delete your browser's [local storage](https://developer.chrome.com/docs/devtools/storage/localstorage/) for the site. There should be a local storage key called `i18nextLng` which you need to delete, after refresh the new language will be detected. This is expected behavior, and it definitely tripped me up when testing after changing my language back and forth. Users don't normally change their language back and forth so it makes sense.
